### PR TITLE
Create mkmf.template.nvhpc

### PR DESCRIPTION
**Pull request by ann-norcio**
## Description:

Created mkmf.template.nvhpc in build_templates to run quickbuild.sh with nvhpc compiler for all models on Derecho. Modified compiler flag `FFLAGS = -O -Mbackslash $(INCS)` to include -Mbackslash, which addresses issue [#555 ](url)

### Fixes issue
Fixes: #554  #555 

### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Documentation changes needed?

- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.

### Tests
Copy mkmf.template.nvhpc and set as your mkmf.template. Set compiler to nvhpc `module load nvhpc`. Then, run quickbuild.sh on DART/models/lorenz_63/work.

## Checklist for merging

- [ ] Updated changelog entry
- [ ] Documentation updated
- [ ] Update conf.py

## Checklist for release
- [ ] Merge into main
- [ ] Create release from the main branch with appropriate tag
- [ ] Delete feature-branch

## Testing Datasets

- [ ] Dataset needed for testing available upon request
- [ ] Dataset download instructions included
- [X] No dataset needed

### CODE REVIEW

**Comment by hkershaw-brown** [comment]
Nice work testing out all the Derecho compilers and finding this, Ann.

There's code in dart that relies on IEEE standard that for a NaN, a /= a is .true.
Background reading if you're interested (#176, #341)

Can you trying building and running the following code on Derecho:

code test_nan.f90
```
program test_nan

implicit none
real :: a, x

x = -1
a = sqrt(x)

print*, a

print*, a /= a
print*, a == a

if (a /= a) then
  print*, "you have a NaN"
endif

end program test_nan
```

You can compile and run it like this:
```
ftn -Kieee test_nan.f90 -o test_nan
./test_nan
```

vs
```
ftn test_nan.f90 -o test_nan
./test_nan
```

We want to make sure a /= a gives .true. for a NaN (we may need to add -Kieee to the compile flags in your mkmf.template.)

And bonus points, can you add some suggested flags if you want to turn on debugging for nvhpc?

**Comment by hkershaw-brown** [comment]
@hkershaw-brown the nan test should go in a developer test, since we are relying on the IEEE behaviour.

File: build_templates/mkmf.template.nvhpc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# If your NETCDF environment variable is not set correctly,
+# uncomment the following line and set value to where lib and include
+# are found for the netcdf files that match this compiler.
+#
+# NETCDF = /opt/local
+
+INCS = -I$(NETCDF)/include
+LIBS = -L$(NETCDF)/lib -lnetcdff -lnetcdf
+FFLAGS = -O -Mbackslash $(INCS)
+LDFLAGS = $(INCS) $(LIBS)
```
Comment (hkershaw-brown) [inline build_templates/mkmf.template.nvhpc:24]:
```suggestion
LDFLAGS = $(INCS) $(LIBS)

# For development or debugging, use the following flags.
# To generate a traceback, 
# set the enviroment variable 'NVCOMPILER_TERM=trace' at runtime.
FFLAGS = -gopt -C -traceback -Ktrap=fp  -Mbackslash -Kieee $(INCS)

```

**Review by hkershaw-brown — CHANGES_REQUESTED** [review]
Looks good Ann, I left a couple of suggestions for ieee and debugging flags.

**Review by mjs2369 — APPROVED** [review]
Tested the compile with the new debugging flags. Approved

**Review by hkershaw-brown — APPROVED** [review]

**Pull request merged**